simpl/schema/external/utils.py

```diff
@@ -89,3 +90,97 @@ def wrapper(*args, **kwargs):
         return func(*args, **kwargs)
 
     return wrapper
+
+
+def walk_fields(info):
+    """
+    Walk all fields in the current info ast.
+    """
+
+    def walk(fields):
+        for field in fields:
+            yield field
+            if isinstance(field, FragmentSpread):
+                selection_set = info.fragments[field.name.value].selection_set
+            else:
+                selection_set = field.selection_set
+            if selection_set:
+                yield from walk(selection_set.selections)
+
+    return walk(info.field_asts)
+
+
+def has_field_named(info, *field_names):
+    """
+    Return true if any fields in the schema match the name(s) provided.
+
+    Use underscored field names and it will look for camelCase versions
+    of those fields too.
+    """
+    assert field_names, (
+        "At least one field name is required "
+        "(remember the first argument should be `info`)."
+    )
```

Comment:
If we want this to raise at runtime, it is good practice to use `raise AssertionError()`. `assert` statements can be disabled.